server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -273,15 +275,20 @@ static ClusterState createDataStream(
             if (isSystem) {
                 throw new IllegalArgumentException("Failure stores are not supported on system data streams");
             }
+            // By hard coding `1` here, we're going to end up with a backing index _and_ a failure index with `generation` equal to 1.
+            // This is not a problem as both have different prefixes (`.ds-` vs `.fs-`) and both will be using the same `generation` field
+            // when rolling over in the future.
             String failureStoreIndexName = DataStream.getDefaultFailureStoreName(dataStreamName, 1, request.getStartTime());
```

Comment:
Nit: Should we extract this to a local variable since it's used in 3 places:
- Failure store name generation
- Backing index name generation
- Data stream creation.

It might help making explicit also that it has these 3 functions.